docker/etc/nginx/custom/03_derivative_images.conf

```diff
@@ -0,0 +1,38 @@
+## Location for derivative images to avoid hitting Drupal for invalid URLs or
+## if the source image doesn't exist.
+location /sites/default/files/styles/ {
+
+  ## Allowed derivative images.
+  ## As mentioned in the main README, this needs to be updated when
+  ## new image styles or different directories to store images are added.
+  location ~ "^/sites/default/files/styles/(announcement|icon|thumbnail|small|medium|large|media_library|guideline_thumbnail)/public/(?<file_path>(images|previews)/[^?]+)" {
```

Comment:
I am tempted to rewrite this (for any sub-directory under `styles/`) so we can ship it in the base php-k8s image and have all Drupals take advantage of it. Apart from the directory names, this is not RW-specific.

Reply:
@cafuegor Yep, I was thinking about that but then that means we lose the implied image style validation and it still becomes possible to hammer Drupal with unwanted requests.

That aside, the rule needs some testing regarding what nginx captures in the `file_path` variable.

On RW9, the files are stored with their UUID so there is no encoding issue but I remember having issues on RW7 with files containing a `#` character for example and so I didn't add such a rule for it.

I will do some tests with various files and let you know how that went.

Reply:
Tested with files containing spaces, `#`, `?` and `%` and that seems to work well. The `$file_path` variable contains the proper url decoded file path.

Created https://github.com/UN-OCHA/docker-images/pull/306 to add the rule to the `php-k8s` images.

Reply:
It would be possible to hammer *nginx* with unwanted requests, but those would never end up hitting Drupal because the validation rule for the source file still need to pass before the request is handed off to Drupal.

Reply:
For example `https://reliefweb.int/sites/default/files/styles/UNKNOWN-IMAGE-STYLE/public/images/reports/f8/cd/f8cdb321-08ff-4b18-9d1e-bc6c608c30f8.png` would be passed to Drupal without the whitelist of image styles because the `/images/reports/f8/cd/f8cdb321-08ff-4b18-9d1e-bc6c608c30f8.png` exists and would result in a Drupal 404 (because the image style doesn't exist).

That said, I think the general rule from https://github.com/UN-OCHA/docker-images/pull/306 is good enough for now as it handles the cases I saw in the logs which were for attachment previews or images of reports that were unpublished or deleted.

So closing this in favor of https://github.com/UN-OCHA/docker-images/pull/306.